pyproject.toml

```diff
@@ -2,5 +2,121 @@
 requires = ["setuptools>=64", "setuptools_scm[toml]>=6.2", "wheel"]
 build-backend = "setuptools.build_meta"
 
+[project]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+description = "One line description of your module"
+dependencies = []
+dynamic = ["version"]
+license = { file = "LICENSE" }
+name = "python3-pip-skeleton"
+readme = "README.rst"
+requires-python = ">=3.8"
+
+[project.optional-dependencies]
+dev = [
+    "black",
+    "isort",
+    "mypy",
+    "flake8",
+    "flake8-isort",
+    "flake8-pyi",
+    "Flake8-pyproject",
+    "pipdeptree",
+    "pre-commit",
+    "pydata-sphinx-theme",
+    "pytest-cov",
+    "setuptools_scm[toml]>=6.2",
+    "sphinx-autobuild",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "tox",
+    "tox-direct",
+    "types-mock",
+]
+
+[project.scripts]
+python3-pip-skeleton = "python3_pip_skeleton.__main__:main"
+
+[project.urls]
+GitHub = "https://github.com/DiamondLightSource/python3-pip-skeleton"
+
+[[project.authors]]
+email = "email@address.com"
+name = "Firstname Lastname"
+
+[[project.maintainers]]
+email = "email@address.com"
+name = "Firstname Lastname"
+
 [tool.setuptools_scm]
 write_to = "src/python3_pip_skeleton/_version.py"
+
+[tool.mypy]
+ignore_missing_imports = true # Ignore missing stubs in imported modules
+
+[tool.isort]
+float_to_top = true
+profile = "black"
+
+[tool.flake8]
+extend-ignore = [
+    "E203", # See https://github.com/PyCQA/pycodestyle/issues/373
+    "F811", # support typing.overload decorator
+    "F722", # allow Annotated[typ, some_func("some string")]
+]
+max-line-length = 88 # Respect black's line length (default 88),
+per-file-ignores = "*pyi:E302"
+exclude = [".tox", "venv"]
+
+
+[tool.pytest.ini_options]
+# Run pytest with all our checkers, and don't spam us with massive tracebacks on error
+addopts = """\
+    --tb=native -vv --doctest-modules --doctest-glob=\"*.rst\"
```

Comment:
I'm unfamiliar with TOML, do we need to escape double quotes within triple quotes?

Reply:
We don't seem to need to, these have now been removed